File: src/hackwars/java/Game/ComputerFunctions/addShowChoices.java
package game.computerfunctions;

import game.*;
import assignments.*;
/**
 A function that is run within Computer.java.

 This function adds to the show choices array. This tells the client to display show choices options.
 */

import java.util.*;

import assignments.*;
import hackscript.model.*;

public class addShowChoices extends function {
    public addShowChoices(Computer MyComputer) {
        super(MyComputer);
    }

    public void execute(ApplicationData MyApplicationData) {
        super.getComputer().sendPacket();
        super.getComputer().sendDamagePacket();
        super.getComputer().getShowChoicesArray().add(MyApplicationData.getParameters());
    }
}

File: src/GameServer/java/com/hackwars/game/functions/AddShowChoices.kt
package com.hackwars.game.functions

import game.ApplicationData
import game.Computer
import game.payload.AddShowChoicesPayload
import game.payloadAs

/**
 * Represents a function that adds a new set of choices to the computer system's "show choices" array.
 * This function operates within the context of the provided application data and modifies the
 * internal state of the associated computer by:
 *
 * - Sending a generic packet to notify other components.
 * - Sending a damage packet to reflect some form of state change.
 * - Adding the application's parameters to the "show choices" array.
 *
 * @param computer The computer instance on which this function operates.
 */
class AddShowChoices(computer: Computer) : Function(computer) {
    override fun execute(applicationData: ApplicationData) {
        val payload = applicationData.payloadAs<AddShowChoicesPayload>()
        computer.sendPacket()
        computer.sendDamagePacket()
        computer.showChoicesArray.add(payload.choices)
    }
}
